ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveyListAdapter.kt

```diff
@@ -42,10 +43,17 @@ class SurveyListAdapter(
 
   /** Binds [SurveyListItem] data to [ViewHolder]. */
   override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-    val item: SurveyListItem = surveys[position]
-    holder.binding.item = item
-    holder.binding.viewModel = viewModel
-    holder.binding.fragment = fragment
+    val surveyListItem = surveys[position]
+    with(holder.binding) {
+      val me = this@SurveyListAdapter
+      item = surveyListItem
+      viewModel = me.viewModel
+      fragment = me.fragment
+
+      val offlineVisibility = if (surveyListItem.availableOffline) View.VISIBLE else View.GONE
+      offlineIcon.visibility = offlineVisibility
+      overflowMenu.visibility = offlineVisibility
+    }
```

Comment:
just a quick guess, but how about if we take the BindingAdapter annotation approach over here?

WDYT?

IMO, onBindViewHolder should be a mediator type, it should not consist the UI logics.

Reply:
We already had the bind adaptor approach which was leading to this problem. Given that we are moving towards jetpack compose, this recycler view / view holder should get replaced soon in the future with the migration.